haystack/document_stores/pinecone.py

```diff
@@ -291,6 +291,13 @@ def get_document_count(
         pinecone_syntax_filter = LogicalFilterClause.parse(filters).convert_to_pinecone() if filters else None
 
         stats = self.pinecone_indexes[index].describe_index_stats(filter=pinecone_syntax_filter)
+
+        if only_documents_without_embedding:
+            count = 0
+            for namespace in stats["namespaces"].keys():
+                if "no-vectors" in namespace:
+                    count += stats["namespaces"][namespace]["vector_count"]
+            return count
         # Document count is total number of vectors across all namespaces (no-vectors + vectors)
         count = 0
         for namespace in stats["namespaces"].keys():
```

Comment:
@NikolaMr seems like a good catch. Couldn't we further simplify the logic just below where we count vectors? I'll let my colleague @bogdankostic chime in here as well.